serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc

```diff
@@ -1 +1,159 @@
+
+[source,json]
+----
+"functionRef": 
+{
+  "refName": "subtraction",
+  "arguments": 
+  {
+    "leftElement": ".fahrenheit",
+    "rightElement": ".subtractValue"
+  }
+}
+----
+
+In the snippet above, we are specifying that the left number of the subtraction is equal to the `fahrenheit` property (which is an input number provided by the user invoking the flow) and that the right number is equal to the `substractvalue` property (which is a constant number injected into the flow model by SetConstants state). After resolving expression evaluation for all properties that contain an expression, the resulting JSON object is passed to the  OpenAPI call. Depending on the OpenAPI definition, properties of that JSON object will be used as body, path, query, or header of the upcoming REST invocation. 
+Note that, for the sake of brevity, the expression is not embedded within `${}`. Kogito will infer it is a valid `jq` expression because of the `.` (period) prefix, but you can embed it if you prefer to do so.
+
+Function arguments might also be defined as a string containing an expression that returns a JSON object.
+[source,json]
+----
+"functionRef": {
+  "refName": "subtraction",
+  "arguments": "{leftElement: .fahrenheit, rightElement : .subtractValue}"
+ }
+
+----
+
+In the snippet above, the whole string is evaluated and the resulting JSON object, which is the same than in the first case, is passed to the OpenAPI call. 
+
+== Data filtering
+
+Serverless workflow specification defines some filtering mechanisms in order to select which data should be part of the flow data model:
+* link:{spec_doc_url}#action-data-filters[action data filter] selects the portion of the action result that will be merged into the model, overriding only those properties in the flow model that share the name with the selected action result
+* link:{spec_doc_url}#event-data-filters[event data filter] same purpose than the previous one, but applied to events rather than actions.
+* link:{spec_doc_url}#state-data-filters[state data filter] sets the whole flow model to the JSON object returned by the expression, discarding any existing property.
+
+We are going to illustrate, using this link:{kogito_sw_examples_url}/serverless-workflow-expression-quarkus[repository], the usage of action and state data filters.
+
+image::core/expression_diagram.png[]
+
+This Serverless Workflow Application accepts as input an array of complex numbers (where `x` is the real coordinate and `y` the imaginary one) and defines an expression function (invoked by `squareState`) to calculate the maximum `x` and the minimum `y` for that `numbers` array. It also contains an action data filter (defined inside `squareState`) that selects the maximum `x` to be merged into the flow model and a state data filter (defined inside `finish` state) that sets that max value as the whole model that will be returned to the caller.
+
+
+.The expression function
```

Comment:
I think this explanation deserves to be on it's own section, the same as you did for the Data Filtering, maybe "Expression Function"

Reply:
I agree that expression functions are quite a simple topic that can be merged in this section.